src/storage/test/ScanEdgeTest.cpp

```diff
@@ -0,0 +1,448 @@
+                    }
+                }
+            }
+            rowNum++;
+        }
+        EXPECT_EQ(expectedRowNum, rowNum);
+        totalRowCount += rowNum;
+    }
+
+    if (resp.has_next) {
+        EXPECT_TRUE(!resp.next_cursor.empty());
+        cursor = resp.next_cursor;
+    } else {
+        EXPECT_TRUE(resp.next_cursor.empty());
+        cursor = "";
+        return;
+    }
+}
+
+TEST(ScanEdgeTest, RetrieveAllPropertyTest) {
+    fs::TempDir rootPath("/tmp/ScanEdgeTest.XXXXXX");
+    std::unique_ptr<kvstore::KVStore> kv = TestUtils::initKV(rootPath.path(), 10);
+
+    LOG(INFO) << "Prepare meta...";
+    auto schemaMan = TestUtils::mockSchemaMan();
+    LOG(INFO) << "Prepare data...";
+    mockData(kv.get());
+    PartitionID partId = 1;
+    auto req = buildRequest(partId, "", 100, true);
+
+    auto* processor = ScanEdgeProcessor::instance(kv.get(), schemaMan.get(), nullptr);
+    auto f = processor->getFuture();
+    processor->process(req);
+    auto resp = std::move(f).get();
+
+    LOG(INFO) << "Check the results...";
+    std::string cursor = "";
+    int32_t rowCount = 0, expectRowCount = 100;
+    checkResponse(partId, resp, cursor, rowCount, expectRowCount, 20, true);
+    EXPECT_EQ(rowCount, 100);
+}
+
+TEST(ScanEdgeTest, RetrieveSomePropertyTest) {
+    fs::TempDir rootPath("/tmp/ScanEdgeTest.XXXXXX");
+    std::unique_ptr<kvstore::KVStore> kv = TestUtils::initKV(rootPath.path(), 10);
+
+    LOG(INFO) << "Prepare meta...";
+    auto schemaMan = TestUtils::mockSchemaMan();
+    LOG(INFO) << "Prepare data...";
+    mockData(kv.get());
+    PartitionID partId = 1;
+    auto req = buildRequest(partId, "", 100);
+
+    auto* processor = ScanEdgeProcessor::instance(kv.get(), schemaMan.get(), nullptr);
+    auto f = processor->getFuture();
+    processor->process(req);
+    auto resp = std::move(f).get();
+
+    LOG(INFO) << "Check the results...";
+    std::string cursor = "";
+    int32_t rowCount = 0, expectRowCount = 100;
+    checkResponse(partId, resp, cursor, rowCount, expectRowCount, 10);
+    EXPECT_EQ(rowCount, 100);
+}
+
+TEST(ScanEdgeTest, RetrieveNonePropertyTest) {
+    fs::TempDir rootPath("/tmp/ScanEdgeTest.XXXXXX");
+    std::unique_ptr<kvstore::KVStore> kv = TestUtils::initKV(rootPath.path(), 10);
+
+    LOG(INFO) << "Prepare meta...";
+    auto schemaMan = TestUtils::mockSchemaMan();
+    LOG(INFO) << "Prepare data...";
+    mockData(kv.get());
+    PartitionID partId = 1;
+    auto req = buildRequest(partId, "", 100, false, true);
+
+    auto* processor = ScanEdgeProcessor::instance(kv.get(), schemaMan.get(), nullptr);
+    auto f = processor->getFuture();
+    processor->process(req);
+    auto resp = std::move(f).get();
+
+    LOG(INFO) << "Check the results...";
+    std::string cursor = "";
+    int32_t rowCount = 0, expectRowCount = 100;
+    checkResponse(partId, resp, cursor, rowCount, expectRowCount, 0, false, true);
+    EXPECT_EQ(rowCount, 100);
```

Comment:
For edges, when no output column attributes are specified. What is output?  
Does it output the following four fields：_src, _rank,  _dst,  _type?

Reply:
We only return the key, aka, src, type and dst (we don't return rank for now)